File: cs/XDust/XDust/IScriptable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace XDust
{
    public interface IScriptable : IEnumerable<String>
    {
        Object Value { get; }
        bool IsValue { get; }
        Object this[String name] { get; }
        bool HasKey(String name);
    }
}

File: java/src/xdust/src/com/heydanno/xdust/IScriptable.java
package com.heydanno.xdust;

import java.lang.reflect.InvocationTargetException;
import java.util.List;

public interface IScriptable extends Iterable<String> {
	
	Object getValue();
	boolean isValue();
	boolean isList();
	boolean isTruthy();
	List<Object> asList();
	Object get(String name) throws IllegalArgumentException, IllegalAccessException, InvocationTargetException;
	boolean hasKey(String name) throws IllegalArgumentException, IllegalAccessException, InvocationTargetException;

}
